File: .tmp_py4013/batch_08_dump/2915_length_of_the_longest_subsequence_that_sums_to_target.js
// LeetCode 2915 - Length of the Longest Subsequence That Sums to Target
// https://leetcode.com/problems/length-of-the-longest-subsequence-that-sums-to-target/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var lengthOfLongestSubsequence = function(nums, target) {
    const dp = Array(target + 1).fill(-1);
    dp[0] = 0;
    for (const v of nums)
        for (let s = target; s >= v; s--)
            if (dp[s - v] >= 0 && dp[s - v] + 1 > dp[s]) dp[s] = dp[s - v] + 1;
    return dp[target];
};

File: .tmp_rs4013/rust_check_12/2915_length_of_the_longest_subsequence_that_sums_to_target.rs
struct Solution;
// LeetCode 2915 - Length of the Longest Subsequence That Sums to Target
// https://leetcode.com/problems/length-of-the-longest-subsequence-that-sums-to-target/

impl Solution {
    pub fn length_of_longest_subsequence(nums: Vec<i32>, target: i32) -> i32 {
        let target = target as usize;
        let mut dp = vec![-1i32; target + 1];
        dp[0] = 0;
        for v in nums {
            let v = v as usize;
            for s in (v..=target).rev() {
                if dp[s - v] >= 0 && dp[s - v] + 1 > dp[s] {
                    dp[s] = dp[s - v] + 1;
                }
            }
        }
        dp[target]
    }
}

fn main() {}
